lib/rules/no-unused-vars.js

```diff
@@ -624,10 +624,26 @@ module.exports = {
 
                     // Report the first declaration.
                     if (unusedVar.defs.length > 0) {
+                        let writeReferences;
+
+                        if (unusedVar.references.length > 0) {
+                            writeReferences = unusedVar.references.filter(ref => ref.isWrite());
+                        }
+
+                        let referenceToReport;
+
+                        if (unusedVar.references.length) {
+
+                            // report last write reference, https://github.com/eslint/eslint/issues/14324
+                            if (Array.isArray(writeReferences) && writeReferences.length) {
+                                referenceToReport = writeReferences[writeReferences.length - 1];
+                            } else {
+                                referenceToReport = unusedVar.references[unusedVar.references.length - 1];
+                            }
```

Comment:
I think we should never report a non-write reference. If there are no write references, then we should report the definition.

For example:

```js
/* eslint no-unused-vars: error */

function fact(n) { // report this `fact`
    if (n < 2) return 1;
    return n * fact(n - 1); // not this
}

```

Reply:
Agreed. That makes more sense.

Reply:
Fixed 👍🏻